scripts/resolve_suggestions_urls.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+import json
+import os
+
+from jsonpath_rw import parse
+from jsonpointer import resolve_pointer, set_pointer
+
+SUGGESTIONS_URL_BASE = "https://cdn.eq.census-gcp.onsdigital.uk/data/v4.0.0"
+
+
+def json_path_to_pointer(json_path):
+    """
+    Convert a json path string into a json pointer string.
+    :param json_path: the input data to search
+    :return: json pointer equivalent to the json path
+    """
+    json_pointer = json_path.replace("[", "").replace("]", "").replace(".", "/")
+    return f"/{json_pointer}"
+
+
+def replace_suggestions_urls(schema_filepath, replacement_url_root):
+    updated_pointer_count = 0
+
+    try:
+        with open(schema_filepath, "r+") as schema_file:
+            schema_json = json.load(schema_file)
+            json_path = parse("$..suggestions_url")
+            for match in json_path.find(schema_json):
+                json_pointer = json_path_to_pointer(str(match.full_path))
+                suggestions_url = match.value
+                if "suggestions_url_root" in suggestions_url:
+                    if replacement_url_root:
+                        replacement_url = suggestions_url.format(
+                            suggestions_url_root=replacement_url_root
+                        )
+                        set_pointer(schema_json, json_pointer, replacement_url)
+                    else:
+                        parent_pointer = "/".join(json_pointer.split("/")[0:-1])
+                        parent_object = resolve_pointer(schema_json, parent_pointer)
+                        del parent_object["suggestions_url"]
+                        set_pointer(schema_json, parent_pointer, parent_object)
+                    updated_pointer_count += 1
+    except FileNotFoundError:
+        print(f"{schema_filepath} not found")
```

Comment:
I think it would make more sense if the file handling code was done outside of this method. As it stands it would output a file not found error, then print how many urls were replaced.